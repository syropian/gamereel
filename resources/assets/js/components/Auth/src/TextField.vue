<template>
<div class="relative">
  <input
    :type="type"
    :name="name"
    :placeholder="placeholder"
    :value="value"
    @input="onInput"
    class="login-field w-full border-2 border-grey-light h-10 px-3 bg-white rounded"
  />
</div>
</template>
<script>
export default {
  props: [
    "type",
    "placeholder",
    "name",
    "value"
  ],
  methods: {
    onInput (e) {
      this.$emit('input', e.target.value)
    }
  }
}
</script>
<style lang="scss">
.login-field {
  transition: border-color 150ms ease;
  &:focus {
    outline: none;
    border-color: config("colors.brand");
  }
}
</style>


