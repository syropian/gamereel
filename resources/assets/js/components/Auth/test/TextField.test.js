import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import TextField from '../TextField'

describe('TextField Component', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(TextField, {
      propsData: {
        name: 'email',
        type: 'email',
        placeholder: 'jane@doe.com',
        value: 'john@doe.com'
      }
    })
  })

  it('passes the props to the input', () => {
    const el = wrapper.find('input').element
    expect(el.getAttribute('name')).toBe('email')
    expect(el.getAttribute('type')).toBe('email')
    expect(el.getAttribute('placeholder')).toBe('jane@doe.com')
    expect(el.getAttribute('value')).toBe(null)
  })

  it('emits an input event on input', () => {
    const input = wrapper.find('input')
    input.element.value = 'Hello'
    input.trigger('input')

    expect(wrapper.emitted().input).toBeTruthy()
    expect(wrapper.emitted().input[0]).toEqual(['Hello'])
  })

  it('matches the correct snapshot', () => {
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
