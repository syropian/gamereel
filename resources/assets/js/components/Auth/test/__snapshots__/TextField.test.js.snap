// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField Component matches the correct snapshot 1`] = `<div class="relative"><input type="email" name="email" placeholder="jane@doe.com" value="john@doe.com" class="login-field w-full border-2 border-grey-light h-10 px-3 bg-white rounded"></div>`;
